### ISSUE TITLE
Distributed tracing page includes ignored route

### ISSUE THREAD

**Issue by aesyondu** [newrelic/newrelic-ruby-agent#853]
Using `newrelic_ignore` on a rails route/controller action still produces distributed tracing as `Controller/Middleware/Rack/ActionDispatch::Routing::RouteSet/call`

## Description
I have rails apps in AWS ECS with load balancer health checking. I added `newrelic_ignore` to the health check route/controller action and it does not show up in the **Summary** and **Transactions** page of the New Relic dashboard. However, it does show up in the **Distributed tracing** page as `Controller/Middleware/Rack/ActionDispatch::Routing::RouteSet/call`

## Expected Behavior
Ignored route/controller action should not show up in the **Distributed tracing** page.

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results
<details>
<summary>
nrdiag (mac)
</summary>

git:(master) ✗ ~/Downloads/nrdiag/mac/nrdiag_x64

Check Results
-------------------------------------------------

Info     Base/Env/CollectEnvVars [Gathered Environment variables of current shell.] 
Success  Base/Config/Collect
Success  Base/Config/Validate
Success  Base/Config/LicenseKey
Success  Base/Config/ValidateLicenseKey
Info     Base/Config/RegionDetect [1 unique New Relic region(s) detected from config.] 
Error    DotNetCore/Env/Versions
Success  Base/Log/Copy
Success  Base/Log/ReportingTo
Info     Base/Containers/DetectDocker [Docker Daemon is Running] 
Success  Ruby/Config/Agent
Info     Ruby/Agent/Version [8.1.0.0, 8.0.0.0] 
Success  Base/Agent/EOL
Success  Base/Collector/ConnectUS
Success  Base/Config/ValidateHSM
Success  Ruby/Env/Process
Info     Ruby/Env/Version [ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64...] 
Success  Ruby/Requirements/Version
Success  Base/Config/AppName
Success  Ruby/Config/Collect
Success  Ruby/Config/IncompatibleGems
Info     Base/Env/HostInfo [Collected host information] 
63 results not shown: 63 None
See nrdiag-output.json for full results.

Issues Found
-------------------------------------------------
Error - DotNetCore/Env/Versions
Unable to complete this health check because we ran into some unexpected errors when attempting to collect this application's .NET Core SDK version:
Unable to run 'dotnet --version':
%w
exec: "dotnet": executable file not found in $PATH

No attachment process specified

For better results, run Diagnostics CLI with the 'suites' option to target a New Relic product. To learn how to use this option, run: '/Users/$USER/Downloads/nrdiag/mac/nrdiag_x64 -h suites'

</details>

## Steps to Reproduce
This sample is created with `rails --minimal --api --skip-bundle newrelic-sample` using ruby@2.6.5 and rails@6.1.4.1

https://github.com/aesyondu/newrelic-sample

1. `vim config/newrelic.yml`
2. input license key
3. `bin/rails s`
4. `curl localhost:3000 # ignored route`
5. `curl localhost:3000/trace_enabled`
6. see in dashboard that the ignored route shows up in distributed tracing as **Controller/Middleware/Rack/ActionDispatch::Routing::RouteSet/call**, along with the trace_enabled route as HealthCheckController#show

<img width="924" alt="image" src="https://user-images.githubusercontent.com/57334943/141399902-b0ec728e-d2ad-4e89-acc9-22d0a3106c31.png">

## Your Environment
Our ecs containers are based on ruby alpine (**ruby:2.6.5-alpine**). However I am able to replicate using local mac:

<details>
<summary>
local mac envinfo
</summary>

  System:
    OS: macOS 10.15.7
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 47.02 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 15.8.0 - ~/.asdf/installs/nodejs/15.8.0/bin/node
    Yarn: 1.22.10 - ~/.asdf/installs/nodejs/15.8.0/.npm/bin/yarn
    npm: 7.5.1 - ~/.asdf/installs/nodejs/15.8.0/bin/npm
  Managers:
    Composer: 2.0.11 - /usr/local/bin/composer
    Homebrew: 3.2.16 - /usr/local/bin/brew
    pip3: 21.1.3 - /usr/local/bin/pip3
    RubyGems: 3.0.3 - /Users/arvinsevilla/.asdf/shims/gem
  Utilities:
    Make: 3.81 - /usr/bin/make
    GCC: 10.15. - /usr/bin/gcc
    Git: 2.30.1 - /usr/local/bin/git
    Clang: 12.0.0 - /usr/bin/clang
    FFmpeg: 4.4 - /usr/local/bin/ffmpeg
  Servers:
    Apache: 2.4.48 - /usr/sbin/apachectl
  Virtualization:
    Docker: 20.10.8 - /usr/local/bin/docker
    VirtualBox: 6.1.18 - /usr/local/bin/vboxmanage
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7199119
    Atom: 1.58.0
    Emacs: 27.2 - /usr/local/bin/emacs
    Nano: 2.0.6 - /usr/bin/nano
    VSCode: 1.62.0 - /usr/local/bin/code
    Vim: 8.2 - /run/current-system/sw/bin/vim
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Bash: 4.4.23 - /run/current-system/sw/bin/bash
    Perl: 5.18.4 - /usr/bin/perl
    PHP: 7.3.29 - /usr/bin/php
    Protoc: 3.15.6 - /usr/local/bin/protoc
    Python: 2.7.16 - /usr/bin/python
    Python3: 3.9.6 - /usr/local/bin/python3
    Ruby: 2.6.5 - /Users/arvinsevilla/.asdf/shims/ruby
  Databases:
    MongoDB: 5.0.2 - /usr/local/bin/mongo
    MySQL: 10.15 - /usr/local/bin/mysql
    SQLite: 3.28.0 - /usr/bin/sqlite3
  Browsers:
    Chrome: 95.0.4638.69
    Edge: 95.0.1020.44
    Firefox: 93.0
    Firefox Developer Edition: 94.0
    Safari: 15.0
    Safari Technology Preview: 14.2

</details>

## Additional context
[According to this post](https://discuss.newrelic.com/t/middleware-call-shown-as-web-transaction/25861/14), moving the `newrelic_rpm` gem to the bottom of the Gemfile fixes the issue. I tried it but the issue still persists (see also sample app linked above).

EDIT: I poured about 4 hours reading the code base. I can't figure out how the distributed tracing data gets passed to the event loop. I'll take a break for now.

**Comment by tannalynn** [comment]
Hello @aesyondu,
Thank you for brining this issue to our attention. This does appear to be a bug in the agent related to span events specifically. 
Currently `newrelic_ignore` appears to be blocking all instrumentation from recording EXCEPT for span events. Span events are also what you see in the distributed tracing view, which explains why you are seeing it there but no where else. You are correct though, that these span events should not be recorded when the transaction is ignored. Our [documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/api-guides/ignoring-specific-transactions/#ignore) also states that events will not be recorded when using `newrelic_ignore`, so this is definitely something we need to fix. 

I investigated this behavior today and I was able to make a PR https://github.com/newrelic/newrelic-ruby-agent/pull/855 to address this issue. I tested it locally and was able to confirm that those changes prevented anything from showing up related to those transactions _including_ in the distributed tracing view. 

Also, I want to thank you for providing that repro you linked to. It is very much appreciated and makes it easier and quicker for me to figure out what the error is! It's a big help.
